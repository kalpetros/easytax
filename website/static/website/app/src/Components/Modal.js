import styles from '../../../css/src/Components/Modal.css';
import {Button} from '../Components/Button';
import {Overlay} from './Overlay';

import React from 'react';

class Modal extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
        };

    }
    componentDidMount() {
    }
    render() {
	return(
            <Overlay>
              <div className={styles.modal}>
                <div className={styles.header}>
                  <span>Add a client</span>
                </div>
                <div className={styles.content}>
                  {this.props.children}
                </div>
                <div className={styles.footer}>
                  <Button title="Create"/>
                </div>
	      </div>
            </Overlay>
	);
    }
}

export{Modal};
