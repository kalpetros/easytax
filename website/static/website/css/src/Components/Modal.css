.modal {
    background-color: #fff;
    max-height: 80vh;
    height: 25vh;
    width: 25vw;
    position: fixed;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    border-radius: 30px;
}
