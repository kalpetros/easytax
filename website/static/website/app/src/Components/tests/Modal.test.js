import {Modal} from '../Modal';

import renderer from 'react-test-renderer';

import React from 'react';

test('Renders a modal', () => {
  const component = renderer.create(
    <Modal>Test</Modal>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
