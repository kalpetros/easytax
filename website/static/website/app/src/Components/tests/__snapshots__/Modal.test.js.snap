// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a modal 1`] = `
<div
  className="overlay"
>
  <div
    className="modal"
  >
    <div
      className="header"
    >
      <span>
        Add a client
      </span>
    </div>
    <div
      className="content"
    >
      Test
    </div>
    <div
      className="footer"
    >
      <button
        className="basic"
      />
    </div>
  </div>
</div>
`;
